### PR TITLE
Add Go code samples

### PR DESCRIPTION
**Pull request by andrii-balitskyi**

### CODE REVIEW

File: src/lib/code-sample/go.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    .map(
+      ([objKey, val]) =>
+        `${pascalCase(objKey)}: ${formatGoValue({ value: val, key: objKey })}`,
+    )
+    .join(', ')
+
+  return `api.${pascalCase(key)}{${formattedEntries}}`
+}
+
+export const createGoResponse = createJsonResponse
```
Comment (andrii-balitskyi) [inline src/lib/code-sample/go.ts:167]: Looks like GO is serializes responses to JSON https://github.com/seamapi/go/blob/e470714a98a413f888a059409a5fd5b9c2383303/core/core.go#L186

Comment (razor-x) [reply src/lib/code-sample/go.ts:167]: It still has to put the data into a Go struct, e.g., https://github.com/seamapi/go/blob/e470714a98a413f888a059409a5fd5b9c2383303/connectwebviews/client.go#L81 https://github.com/seamapi/go/blob/e470714a98a413f888a059409a5fd5b9c2383303/connect_webviews.go#L246

See https://drstearns.github.io/tutorials/gojson/

Basically, Go has some syntax to make parsing JSON into structs simpler, but in the GO code, you still interact with the data as a struct, e.g.,

```go
    fmt.Printf(
        "%s",
        Device.ConnectedAccountID,
    )

File: src/lib/code-sample/go.ts [hunk 2]
```diff
@@ -0,0 +1,167 @@
+import { pascalCase } from 'change-case'
+
+import type { Json, NonNullJson } from 'lib/json.js'
+
+import { createJsonResponse } from './create-json-response.js'
+import type { CodeSampleDefinition, Context } from './schema.js'
+
+const DEFAULT_GO_PACKAGE_NAME = 'api'
+const GO_PACKAGE_BASE_PATH = 'github.com/seamapi/go'
```
Comment (razor-x) [inline src/lib/code-sample/go.ts:9]:
```suggestion
const defaultGoPackageName = 'api'
const goPackageBasePath = 'github.com/seamapi/go'
```

**Comment by razor-x** [comment]
@andrii-balitskyi Can you update from main?

File: src/lib/code-sample/go.ts (hunk 1 above)
Comment (razor-x) [reply src/lib/code-sample/go.ts:167]: We can merge this with the json response type for now, and update it to `go` in another PR.

File: src/lib/code-sample/go.ts [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+  const pathParts = request.path.split('/')
+
+  return `${goSdkImports}
+
+  client${pathParts.map((p) => pascalCase(p)).join('.')}(${goSdkRequestArgs})
+  `.trim()
+}
+
+const getGoPackageName = (path: string): string => {
```
Comment (razor-x) [inline src/lib/code-sample/go.ts:35]: What is an example where the import is not `import api "github.com/seamapi/go`?

File: src/lib/code-sample/go.ts (hunk 2 above)
Comment (razor-x) [reply src/lib/code-sample/go.ts:9]: We should actually make these config options for blueprint. But ok as consts now.

**Review by razor-x — APPROVED** [review]

File: src/lib/code-sample/go.ts (hunk 3 above)
Comment (andrii-balitskyi) [reply src/lib/code-sample/go.ts:35]: For ex., `import acs "github.com/seamapi/go/acs` for `acs.UsersListAccessibleEntrancesRequest` or 
`import accesscodes "github.com/seamapi/go/accesscodes` for `accesscodes.UnmanagedConvertToManagedRequest` or
`import useridentities "github.com/seamapi/go/useridentities` for `useridentities`EnrollmentAutomationsDeleteRequest` etc.

File: src/lib/code-sample/go.ts (hunk 1 above)
Comment (andrii-balitskyi) [reply src/lib/code-sample/go.ts:167]: https://github.com/seamapi/blueprint/issues/97

**Pull request merged**